onnx/shape_inference.py

```diff
@@ -88,16 +88,13 @@ def infer_node_outputs(
     input_types: Dict[str, onnx.TypeProto],
     input_data: Optional[Dict[str, onnx.TensorProto]] = None,
     input_sparse_data: Optional[Dict[str, onnx.SparseTensorProto]] = None,
-    outer_scope_value_types: Optional[Dict[str, onnx.TypeProto]] = None,
 ) -> Dict[str, onnx.TypeProto]:
     if not schema.has_type_and_shape_inference_function:  # type: ignore
         return {}
     if input_data is None:
         input_data = {}
     if input_sparse_data is None:
         input_sparse_data = {}
-    if outer_scope_value_types is None:
-        outer_scope_value_types = {}
 
     # To avoid copying on C++ side, pass only what is needed for this inference call
     passed_input_types = {
```

Comment:
Currently, the filtering in the following two lines is too aggressive (since we are using this for outer-scope values as well). So, the filtering in the lines below has to be removed.

Reply:
So, I suggest `passed_input_types = input_types`